packages/core/src/components/dialog/multistepDialog.tsx

```diff
@@ -0,0 +1,236 @@
+import { IStepProps, Step, StepId } from "./step";
+
+export type IFinalButtonProps = Pick<IButtonProps, "active" | "loading" | "rightIcon"> &
+    IActionProps &
+    IElementRefProps<any>;
+
+type StepElement = React.ReactElement<IStepProps & { children: React.ReactNode }>;
+
+export interface IMultistepDialogProps extends IDialogProps {
+    /**
+     * Props for the button to display on the final step. To illustrate, this may be a submit button.
+     */
+    finalButtonProps?: IFinalButtonProps;
+
+    /**
+     * A callback that is invoked when the user selects a different step by clicking on back, next, or a step itself.
+     */
+    onChange?(newStepId: StepId, prevStepId: StepId | undefined, event: React.MouseEvent<HTMLElement>): void;
+}
+
+interface IMultistepDialogState {
+    enableNextButton: boolean;
+    lastViewedIndex: number;
+    selectedIndex: number;
+}
+
+const PADDING_BOTTOM = 0;
+
+const MIN_WIDTH = 800;
+
+@polyfill
+export class MultistepDialog extends AbstractPureComponent2<IMultistepDialogProps, IMultistepDialogState> {
+    public static displayName = `${DISPLAYNAME_PREFIX}.MultistepDialog`;
+
+    public static defaultProps: Partial<IMultistepDialogProps> = {
+        canOutsideClickClose: true,
+        isOpen: false,
+    };
+
+    public constructor(props: IMultistepDialogProps) {
+        super(props);
+        this.state = this.getDefaultState();
+    }
+
+    public render() {
+        return (
+            <Dialog {...this.props} style={this.getDialogStyle()}>
+                <div className={Classes.MULTISTEP_DIALOG_PANELS}>
+                    {this.renderLeftPanel()}
+                    {this.maybeRenderRightPanel()}
+                </div>
+            </Dialog>
+        );
+    }
+
+    public componentDidUpdate(prevProps: IMultistepDialogProps) {
+        if (!prevProps.isOpen && this.props.isOpen) {
+            this.setState(this.getDefaultState());
+        }
+    }
+
+    private getDefaultState() {
+        const steps = this.getStepChildren();
+        return {
+            enableNextButton: steps.length > 0 ? isNextButtonEnabled(steps[0]) : false,
+            lastViewedIndex: 0,
+            selectedIndex: 0,
+        };
+    }
+
+    private getDialogStyle() {
+        return { paddingBottom: PADDING_BOTTOM, minWidth: MIN_WIDTH, ...this.props.style };
+    }
+
+    private renderLeftPanel() {
+        return (
+            <div className={Classes.MULTISTEP_DIALOG_LEFT_PANEL}>
+                {React.Children.toArray(this.props.children).filter(isStepElement).map(this.renderStep)}
+            </div>
+        );
+    }
+
+    private renderStep = (step: StepElement, index: number) => {
+        const stepNumber = index + 1;
+        const hasBeenViewed = this.state.lastViewedIndex >= index;
+        const currentlySelected = this.state.selectedIndex === index;
+        return (
+            <div className={classNames(Classes.STEP_CONTAINER, hasBeenViewed && Classes.ACTIVE)} key={index}>
```

Comment:
```suggestion
            <div className={classNames(Classes.STEP_CONTAINER, { [Classes.ACTIVE]: hasBeenViewed })} key={index}>
```
Looks like other places in BP use this syntax for conditional classes but defer to @adidahiya